sql/parse/parse.go

```diff
@@ -2423,6 +2430,147 @@ func havingToHaving(ctx *sql.Context, having *sqlparser.Where, node sql.Node) (s
 	return plan.NewHaving(cond, node), nil
 }
 
+// windowToWindow will apply a WINDOW clause to a plan.Window node,
+// replacing named window references with their window definitions.
+// Performs semantic validation of window compatibility/validity.
+// TODO: separate plan building from replacement and semantic validation
+func windowToWindow(ctx *sql.Context, window sqlparser.Window, node sql.Node) (sql.Node, error) {
```

Comment:
This is definitely pretty gross, we don't want the parser doing this much work. Parsing should be a strictly mechanical process, no logic. Need to correct this on a future pass before it stagnates.